test/boost/sstable_3_x_test.cc

```diff
@@ -4586,8 +4586,8 @@ static std::unique_ptr<index_reader> get_index_reader(shared_sstable sst, reader
                                           tracing::trace_state_ptr(), use_caching::yes);
 }
 
-shared_sstable make_test_sstable(test_env& env, schema_ptr schema, const sstring& table_name, sstables::generation_type::int_t gen = 1) {
-    return env.reusable_sst(schema, get_read_index_test_path(table_name), gen).get0();
+shared_sstable make_test_sstable(test_env& env, schema_ptr schema, const sstring& table_name) {
+    return env.reusable_sst(schema, get_read_index_test_path(table_name), sstables::generation_type{1}).get0();
```

Comment:
We could even get rid of this `sstables::generation_type{1}`, by adding a `sstables::generation_type::make-random_generation()`, or an equivalent function. It is clear that this caller doesn't care what the generation is, it just needs a valid value.

Reply:
@denesb i had a hard time this morning to understand the test failure from `test_sstable_move_exists_failure` with my change. and i ended up realizing that this test was actually using static data hosted by this source repo for testing, see https://github.com/scylladb/scylladb/tree/master/test/resource/sstables/compressed . so i reverted the change using generation_generator in this very test and stick to the hard coded values. i am not happy with the way how i fixed the test failure. but i think i need to get a compromise between a satisfying fix and a tangible progress in the UUID generation identifier work.

so, again, let's get this done piece meal.